src/odemis/gui/cont/tabs/sparc2_align_tab.py

```diff
@@ -2037,6 +2054,26 @@ def _on_spec_ded_aligner_init_move(self, f: Future) -> None:
         except Exception as ex:
             logging.error("spec-ded-aligner initialisation move failed: %s", ex)
 
+    def _on_spec_ded_aligner_pos(self, pos):
+        """
+        Called when the spec-ded-aligner is moved (and the tunnel align mode is active),
+          for updating the MD_FAV_POS_ACTIVE
+        """
+        if self.tab_data_model.main.tab.value != self:
+            # Should never happen, but for safety, we double check
+            logging.warning("Received active spec-ded-aligner position while outside of alignment tab")
+            return
+
+        # Update the axis which are supposed to be recorded: the ones already in MD_FAV_POS_ACTIVE
+        aligner = self.tab_data_model.main.spec_ded_aligner
```

Comment:
According to the docstring, should the code not be in mode == "tunnel-lens-align":

Reply:
Yes, that's what `_on_tunnel_lens_align_done()` does: it activates this function (ie, subscribe to the position updates of the spec_ded_aligner), only when tunel-lens-align is active.